packages/slate-editor/src/modules/editor-v4-components/BookmarkCard/BookmarkCard.stories.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+
+import * as BookmarkCard from './index';
+
+export default {
+    title: 'Modules/Components/BookmarkCard',
+    argTypes: {
+        showUrl: { control: 'boolean' },
+    },
+};
+
+const info = {
+    url: 'https://www.prezly.com/index',
+    title: 'PR Software for better, faster PR',
+    description:
+        'Prezly provides software for communications teams that want to do more. Manage your media relations, create online newsrooms, and publish and pitch your stories from one place.',
+    provider_name: 'Prezly.com',
+    thumbnail_url:
+        'https://cdn.uc.assets.prezly.com/ab836005-6e93-4c98-bbca-a39cf50e0a78/twitter_card.png',
+    thumbnail_width: 1500,
+    thumbnail_height: 500,
+};
+
+export function Horizontal(props: { showUrl: boolean }) {
+    return (
+        <div style={{ width: 680 }}>
+            <BookmarkCard.Container isSelected layout="horizontal">
+                <BookmarkCard.Thumbnail
+                    href={info.url}
+                    src={info.thumbnail_url}
+                    width={info.thumbnail_width}
+                    height={info.thumbnail_height}
+                />
+                <BookmarkCard.Details
+                    href={info.url}
+                    title={info.title}
+                    description={info.description}
+                >
+                    <BookmarkCard.Provider
+                        showUrl={props.showUrl}
+                        url={info.url}
+                        providerName={info.provider_name}
+                    />
+                </BookmarkCard.Details>
+            </BookmarkCard.Container>
```

Comment:
Did you consider making the `BookmarkCard` component a black box where everything is passed inside as properties? (in contrast to building a card with sub-components)

IMO it's just easier to use this way and less code duplication.

```tsx
<BookmarkCard
   url={...}
   title={...}
   description={...}
   providerName={...}
   thumbnailUrl={...}
   thumbnailWidth={...}
   thumbnailHeight={...}
/>
```

Reply:
There can be conditions in rendering different parts of the card, or if you need to wrap some nodes with other elements, or if you need to add some custom elements inside the card. Thus you have to add new props to cover all these needs

Reply:
Are you saying that this *might possibly* be needed in a future use-case that does not exist yet? If so, I'd suggest to keep things simple now and complicate them only when there is a need for that.

Reply:
We already have conditions in JSX
personally i would go with composition if it is possible. Thus in JSX you see where the props are bypassed rather than see a component with a bunch of props with layouting like condition to render this or that component

Reply:
I don't share your point of view. But alright, I won't continue this discussion. Let's do it this way, with composition.